beaconChain/node/index.ts

```diff
@@ -0,0 +1,27 @@
+import { DEPOSIT_CONTRACT_ADDRESS } from "../constants/constants";
+import { BeaconState } from "../interfaces/state";
+import { getInitialBeaconState } from "../lib/state/index";
+import {getInitialDeposits} from "../pow/fake";
+
+interface BeaconNodeCtx {
+  chain: string;
+  state: BeaconState;
+  powChainAddress: string;
+}
+
+class BeaconNode {
+  public chain: string;
+  public state: BeaconState;
+  public powChainAddress: string;
+
+  constructor(opts: BeaconNodeCtx) {
+    this.chain = opts.chain ? opts.chain : "mainnet";
+    this.powChainAddress = opts.powChainAddress ? opts.powChainAddress : DEPOSIT_CONTRACT_ADDRESS;
+
+    // TODO :: Replace stubbed functionality
+    const initialDeposits = getInitialDeposits(); // Stubbed - uses fake data
```

Comment:
I think this is good for now to make forward progress.
We should remember that in the final version, according to the spec, the beacon chain state starts after a `ChainStart` log gets emitted. In our codebase, this may look like starting a `BeaconNode`, which then waits to actually start, triggering `getInitialBeaconState` and other things, after it gets the appropriate log.

Reply:
This is a good point. We can recycle this to a different function. 

I sort of like how prysm registers a bunch of "services" so maybe we can do that in the constructor instead...d